controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -39,17 +59,55 @@ int main(int argc, char** argv) {
     }
     // Load def argument which will be used to identify the soccer field using the webots getFromDef function
     std::string fieldDef = argv[1];
+    // Load def argument which will be used to identify the current robot using the webots getFromDef function
+    std::string selfDef = argv[2];
 
     // Load def arguments of other robots in the field into a vector
-    std::vector<webots::Node*> otherRobotsNodes;
+    std::vector<webots::Node*> robotsNodes;
     // Only one supervisor instance can exist in a single controller. As such the same one will be used
     // every time
     webots::Supervisor supervisor = webots::Supervisor();
     for (int i = 2; i < argc; i++) {
         // Load the Node of the robot by def argument and add to vector
-        otherRobotsNodes.emplace_back(supervisor.getFromDef(argv[i]));
+        robotsNodes.emplace_back(supervisor.getFromDef(argv[i]));
     }
 
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
```

Comment:
Should do consistent casing in here, there's a big mix of camelCase and snake_case.